source/io/module_eggShell.js

```diff
@@ -205,7 +214,7 @@
 
         Module.eggShell.createValueRef = function (typeRef, dataRef) {
             if (typeof typeRef !== 'number' || typeof dataRef !== 'number' ||
-                typeRef <= 0 || dataRef <= 0) {
+                (typeRef <= 0 && dataRef <= 0)) {
```

Comment:
@gleonoliva @rajsite : The reason I changed this is the following: There are valid situations where typeRef is not null and valueRef can be null (for example, when wild card argument is passed for a function in VIA). After createValueRef() is called, I would like to still be able to use valueRef.typeRef in those situations without exceptions thrown.

Reply:
OTOH, if dataRef is nulll in Vireo (0 here in JS), we could set dataRef to undefined and leave typeRef to contain valid value. Would you rather prefer that as opposed to leaving dataRef as 0?

Reply:
Sankara, both appveyor and travis-ci failed. We need to look into these errors.

Reply:
Of the options described the second one sounds better. Hard check that typeRef <=0 but when dataRef <= 0 set the valueRef.dataRef to undefined.

If we don't leak DOM "null" values in ASW we shouldn't leak Vireo null values (zeroes) in Travis. Coerce nulls to undefined.

---

Out of curiosity, what is the situation where you need to know the type of a unwired input? Why does this need to loosen assertions to account for this case?

Reply:
I think I found the situation below, and my comments down there make me think we should not loosen the check here.

Reply:
I think I disagree here. If we keep the behavior in `createValueRef` to return `undefined`. We can rewrite those two other places like:
```
var updateReturnValue = function (functionName, returnValueRef, returnUserValue, errorValueRef) {
if (returnValueRef === undefined) {
//............
}
```

and

```
Module.javaScriptInvoke.jsJavaScriptInvoke = function (
...............
var shouldReturnValueBeUpdated = shouldUpdateReturnValue(isInternalFunction, returnValueRef);
}
// Where ...

var shouldUpdateReturnValue = function (isInternalFunction, returnValueRef) {
if (returnValueRef === undefined) return false;
...
```
So far we've been guarding us against those cases where one of them is valid and the other is not. 
I don't see a strong reason to loosen that check in `createValueRef`. Am I missing something?

Reply:
I cleaned up this area of code after exposing read/write Javascript in eggshell, so what I was doing was no longer required.